alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/conditions/TensionPresent.java

```diff
@@ -0,0 +1,105 @@
+package it.unibo.alchemist.model.implementations.conditions;
+
+import java.util.stream.Stream;
+
+import it.unibo.alchemist.model.interfaces.CellWithCircularArea;
+import it.unibo.alchemist.model.interfaces.CircularDeformableCell;
+import it.unibo.alchemist.model.interfaces.Context;
+import it.unibo.alchemist.model.interfaces.Environment;
+import it.unibo.alchemist.model.interfaces.EnvironmentSupportingDeformableCells;
+import it.unibo.alchemist.model.interfaces.Node;
+
+/**
+ * 
+ */
+public class TensionPresent extends AbstractCondition<Double> {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = 1L;
+    private final EnvironmentSupportingDeformableCells env;
+
+    /**
+     * 
+     * @param node 
+     * @param env 
+     */
+    public TensionPresent(final Environment<Double> env, final Node<Double> node) {
+        super(node);
+        if (node instanceof CircularDeformableCell) {
+            if (env instanceof EnvironmentSupportingDeformableCells) {
+                this.env = (EnvironmentSupportingDeformableCells) env;
+            } else {
+                throw new IllegalArgumentException("This Condition can only be supported in an EnironmentSupportingDeformableCells");
+            }
+        } else {
+            throw new IllegalArgumentException("This Condition can only be setted in a CircularDeformableCell");
+        } 
+    }
+
+    @Override
+    public TensionPresent cloneOnNewNode(final Node<Double> n) {
+        return new TensionPresent(env, n);
+    }
+
+    @Override
+    public Context getContext() {
+        return Context.NEIGHBORHOOD;
+    }
+
+    @Override
+    public double getPropensityConditioning() {
+        final CircularDeformableCell thisNode = (CircularDeformableCell) getNode();
+        return env.getNodesWithinRange(thisNode, env.getMaxDiameterAmongDeformableCells()).stream()
+                //.parallel()
+                .flatMap(n -> n instanceof CellWithCircularArea 
+                        ? Stream.of((CellWithCircularArea) n) 
+                                : Stream.empty())
+                .mapToDouble(n -> {
+                    final double maxRn;
+                    final double minRn;
+                    final double maxRN = thisNode.getMaxRadius();
+                    final double minRN = thisNode.getRadius();
+                    if (n instanceof CircularDeformableCell) {
```

Comment:
Cast once and save:
final CircularDeformableCell cell = (CircularDeformableCell) n